ambry-tools/src/main/java/com.github.ambry/store/DumpDataTool.java

```diff
@@ -0,0 +1,309 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.ClusterMapManager;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.Config;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.messageformat.MessageFormatException;
+import com.github.ambry.utils.Utils;
+import java.io.EOFException;
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import org.json.JSONException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.store.PersistentIndex.*;
+
+
+/**
+ * Tool to assist in dumping data from data files in Ambry
+ * Supported operations are
+ * 1. Compare Index entries to Log entries
+ * 2. Compare all entries in all indexes in a replica to Log entries
+ * 3. Compare Log entries to index entries
+ */
+public class DumpDataTool {
+  private final ClusterMap clusterMap;
+  // The index file that needs to be dumped for comparison purposes
+  @Config("file.to.read")
+  private final String fileToRead;
+
+  /**
+   * File path referring to the hardware layout
+   */
+  @Config("hardware.layout.file.path")
```

Comment:
not sure if config decorations make sense for private variables (here and in consistency checker)